process/physics.py

```diff
@@ -379,389 +498,966 @@ def conhas(alphaj, alphap, bt, delta95, eps, kappa95, p0, rmu0):
+    tempe: np.ndarray,
+    tempi: np.ndarray,
+    inverse_q: np.ndarray,
+    rho: np.ndarray,
+) -> np.ndarray:
+    """
+    Calculate the local beta poloidal including ion pressure for the Sauter bootstrap current scaling.
+
+    Parameters:
+    - radial_elements: np.ndarray, radial element indexes in range 1 to nr
+    - nr: int, maximum value of radial_elements
+    - rmajor: float, plasma major radius (m)
+    - bt: float, toroidal field on axis (T)
+    - ne: np.ndarray, electron density profile (/m^3)
+    - ni: np.ndarray, ion density profile (/m^3)
+    - tempe: np.ndarray, electron temperature profile (keV)
+    - tempi: np.ndarray, ion temperature profile (keV)
+    - inverse_q: np.ndarray, inverse safety factor profile
+    - rho: np.ndarray, normalized minor radius profile
+
+    Returns:
+    - np.ndarray, the local total beta poloidal
+
+    Reference:
+    - None
     """
-
     return (
         np.where(
-            j != nr,
+            radial_elements != nr,
             1.6e-4
             * np.pi
             * (
-                ((ne[j] + ne[j - 1]) * (tempe[j] + tempe[j - 1]))
-                + ((ni[j] + ni[j - 1]) * (tempi[j] + tempi[j - 1]))
+                (
+                    (ne[radial_elements] + ne[radial_elements - 1])
+                    * (tempe[radial_elements] + tempe[radial_elements - 1])
+                )
+                + (
+                    (ni[radial_elements] + ni[radial_elements - 1])
+                    * (tempi[radial_elements] + tempi[radial_elements - 1])
+                )
+            ),
+            6.4e-4
+            * np.pi
+            * (
+                ne[radial_elements - 1] * tempe[radial_elements - 1]
+                + ni[radial_elements - 1] * tempi[radial_elements - 1]
             ),
-            6.4e-4 * np.pi * (ne[j - 1] * tempe[j - 1] + ni[j - 1] * tempi[j - 1]),
         )
-        * (rmajor / (bt * rho[j - 1] * np.abs(mu[j - 1] + 1.0e-4))) ** 2
+        * (
+            rmajor
+            / (
+                bt
+                * rho[radial_elements - 1]
+                * np.abs(inverse_q[radial_elements - 1] + 1.0e-4)
+            )
+        )
+        ** 2
     )
 
 
 @nb.jit(nopython=True, cache=True)
-def tpf(j, triang, sqeps, fit=1):
-    """Trapped particle fraction
-    author: P J Knight, CCFE, Culham Science Centre
-    j  : input integer : radial element index in range 1 to nr
-    fit : input integer : (1)=ASTRA method, 2=Equation from Sauter2002, 3=Equation from Sauter2013
-    This function calculates the trapped particle fraction at
-    a given radius.
-    <P>A number of different fits are provided, but the one
-    to be used is hardwired prior to run-time.
-    <P>The code was supplied by Emiliano Fable, IPP Garching
-    (private communication).
-    O. Sauter et al, Plasma Phys. Contr. Fusion <B>44</B> (2002) 1999
-    O. Sauter, 2013:
-    http://infoscience.epfl.ch/record/187521/files/lrp_012013.pdf
+def trapped_particle_fraction(
```

Comment:
should be 'hidden'?

And should the name make it clear its for the Sauter scaling?

Reply:
Re-named